hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/keygen/TestTimestampBasedKeyGenerator.java

```diff
@@ -223,9 +223,9 @@ public void test_ExpectsMatch_SingleInputFormat_ISO8601WithMsZ_OutputTimezoneAsU
     baseRecord.put("createTime", "2020-04-01T13:01:33.428Z");
     properties = this.getBaseKeyConfig(
         "DATE_STRING",
-        "yyyy-MM-dd'T'HH:mm:ss.SSSZ",
-        "",
+        "yyyy-MM-dd'T'HH:mm:ss.SSSX",
         "",
+        "GMT",
```

Comment:
as commented above, since using java time will by default give system default timezone, we need to set the timezone specifically to make sure it matches with input data's timezone `Z`